src/Controls/GlobeControls.js

```diff
@@ -521,60 +542,145 @@ class GlobeControls extends THREE.EventDispatcher {
 
-        event.preventDefault();
-        const coords = this.view.eventToViewCoords(event);
+        rotateEnd.copy(event.viewCoords);
+        rotateDelta.subVectors(rotateEnd, rotateStart);
 
-        switch (this.state) {
-            case this.states.ORBIT:
-            case this.states.PANORAMIC: {
-                rotateEnd.copy(coords);
-                rotateDelta.subVectors(rotateEnd, rotateStart);
+        const gfx = this.view.mainLoop.gfxEngine;
 
-                const gfx = this.view.mainLoop.gfxEngine;
-                this.rotateLeft(2 * Math.PI * rotateDelta.x / gfx.width * this.rotateSpeed);
-                // rotating up and down along whole screen attempts to go 360, but limited to 180
-                this.rotateUp(2 * Math.PI * rotateDelta.y / gfx.height * this.rotateSpeed);
+        sphericalDelta.theta -= 2 * Math.PI * rotateDelta.x / gfx.width * this.rotateSpeed;
+        // rotating up and down along whole screen attempts to go 360, but limited to 180
+        sphericalDelta.phi -= 2 * Math.PI * rotateDelta.y / gfx.height * this.rotateSpeed;
 
-                rotateStart.copy(rotateEnd);
-                break; }
-            case this.states.DOLLY:
-                dollyEnd.copy(coords);
-                dollyDelta.subVectors(dollyEnd, dollyStart);
+        rotateStart.copy(rotateEnd);
+        this.update();
+    }
 
-                this.dolly(-dollyDelta.y);
-                dollyStart.copy(dollyEnd);
-                break;
-            case this.states.PAN:
-                panEnd.copy(coords);
-                panDelta.subVectors(panEnd, panStart);
+    initDrag(event) {
+        this.initMovement();
 
-                this.mouseToPan(panDelta.x, panDelta.y);
+        if (this.view.getPickingPositionFromDepth(event.viewCoords, pickingPoint)) {
+            this.state = this.states.MOVE_GLOBE;
+            pickSphere.radius = pickingPoint.length();
+            lastNormalizedIntersection.copy(pickingPoint).normalize();
+            this.updateHelper(pickingPoint, helpers.picking);
+        }
+    }
 
-                panStart.copy(panEnd);
-                break;
-            case this.states.MOVE_GLOBE: {
-                const normalized = this.view.viewToNormalizedCoords(coords);
-                raycaster.setFromCamera(normalized, this.camera);
-                // If there's intersection then move globe else we stop the move
-                if (raycaster.ray.intersectSphere(pickSphere, intersection)) {
-                    normalizedIntersection.copy(intersection).normalize();
-                    moveAroundGlobe.setFromUnitVectors(normalizedIntersection, lastNormalizedIntersection);
-                    lastTimeMouseMove = Date.now();
-                } else {
-                    this.onMouseUp();
-                }
-                break; }
-            default:
+    handleDrag(event) {
+        if (this.player.isPlaying()) {
+            this.player.stop();
```

Comment:
may be is unnecessary to stop player, WDYT?

Reply:
Indeed, the only case where this is needed is with rotation, as damping can occur while moving the mouse (without releasing its button). When not checked in `handleRotation` the rotation movement can be a little bit jerky. Therefore, I left these lines in `handleRotation` and removed it from other movement methods.